pkg/clients/transportclient/transportclient_impl.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package transportclient
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"google.golang.org/grpc/connectivity"
+
+	"google.golang.org/grpc"
+)
+
+// Dial using the authenticated mode
+func (o *transportClient) DialNode(ctx context.Context, addr string) (conn *grpc.ClientConn, err error) {
+	/* TODO@ASK security feature under development */
+	return o.DialUnauthenticated(ctx, addr)
+}
+
+// Dial using unauthenticated mode
+func (o *transportClient) DialUnauthenticated(ctx context.Context, addr string) (conn *grpc.ClientConn, err error) {
+	/* A dozen attempts... Recommendation: this value should be configurable */
+	maxAttempts := 12
+	conn, err = grpc.Dial(addr, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+
+	/* connection retry attempt */
+	for (conn.GetState() != connectivity.State(connectivity.Ready)) && ((maxAttempts <= 12) && (maxAttempts > 0)) {
```

Comment:
does this really work?

Reply:
i'd expect to see new attempts to redial.

that said, i'm not sure we need retry attempts yet

Reply:
removed the error/retry attempt code